Code-Generation/Sources/generate-swiftbasicformat/GenerateSwiftBasicFormat.swift

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+/// SwiftSyntaxBuilder sources to be generated
+private let sourceTemplates: [(SourceFile, String)] = [
+  (formatFile, "Format.swift"),
+]
+
+@main
+struct GenerateSwiftSyntaxBuilder: ParsableCommand {
+  @Argument(help: "The path to the destination directory where the source files are to be generated")
+  var generatedPath: String
+
+  @Flag(help: "Enable verbose output")
+  var verbose: Bool = false
+
+  func run() throws {
+    let generatedURL = URL(fileURLWithPath: generatedPath)
+    let format = Format(indentWidth: 2)
+
+    try FileManager.default.createDirectory(
+      atPath: generatedURL.path,
+      withIntermediateDirectories: true,
+      attributes: nil
+    )
+
+    for (sourceFile, name) in sourceTemplates {
```

Comment:
Perhaps we could move all of this logic from `main` into a function taking the generation path/url and a `[(SourceFile, String)]` and share it with `generate-swiftsyntaxbuilder`?

Reply:
Oh, I wanted to do that but forgot about it while I was working on this PR. Moved it to `Utils` now. Thanks!